src/manager/k8s.service.ts

```diff
@@ -1,24 +1,193 @@
 import { Injectable } from '@nestjs/common';
 import { IAgentManager } from './agent.manager.interface';
 import { AgentConfig } from './agent.config';
+import { KubeConfig, CoreV1Api, V1Secret, V1Deployment, V1Pod, V1Service, V1PodCondition } from '@kubernetes/client-node';
+import { Logger } from 'protocol-common/logger';
+import { readFile } from 'fs';
+import cryptoRandomString from 'crypto-random-string';
 
 /**
  * Starts and stops agents within a kubernetes environment
  */
 @Injectable()
 export class K8sService implements IAgentManager {
 
+    // newKubeClient
+    // This configuration is only available if you run the server on a filesystem provisioned with a k8s client.
+    // When running in k8s, this config will be available via the configured k8s service account of the pod.
+    // Locally, this config is provided via ~/.kube/config.
+    newKubeClient = (): CoreV1Api => {
+        const kc = new KubeConfig();
+        kc.loadFromDefault();
+        const k = kc.makeApiClient(CoreV1Api);
+        return k;
+    }
+    kapi = this.newKubeClient();
+
+    // getNamespace
+    // This will work in k8s when the service account is set up correctly.
+    // Locally this will fail, unless you create this file at the correct path.
+    getNamespace = (): string => {
+      let namespace = 'default';
+      readFile('/var/run/secrets/kubernetes.io/serviceaccount/namespace','utf8', (err,data) => {
+        if (err) throw err;
+        namespace = data;
+      });
+      return namespace;
+    }
+    namespace = this.getNamespace();
```

Comment:
lets change class members to match our uses in other classes.  https://github.com/kiva/aries-guardianship-agency/blob/master/src/manager/agent.manager.service.ts